js/ui/control/attribution.js

```diff
@@ -67,8 +67,14 @@ AttributionControl.prototype = util.inherit(Control, {
             }
             return true;
         });
-
         this._container.innerHTML = attributions.join(' | ');
+
+        var attributionNodes = this._container.childNodes;
+        for (var i = 0; i < attributionNodes.length; i++) {
+            if (attributionNodes[i].classList && attributionNodes[i].classList.contains('mapbox-improve-map')) {
+                this._editLink = attributionNodes[i];
+            }
+        }
```

Comment:
If I put the logic in `_updateEditLink` the node is not attached the document and so changes to the node's `href` property do not update on the live attribution node. I had `document.getElementsByClassName` and that was failing the tests, but I didn't realize I could do `this._container.getElementsByClassName` 😮  I'll change that now.